src/ophyd_async/core/device.py

```diff
@@ -52,6 +60,9 @@ def set_name(self, name: str):
         name:
             New name to set
         """
+        if hasattr(self, "log"):
+            del self.log
```

Comment:
A comment as to intent might be good here
```suggestion
        # If changing name, delete self.log so a new one with the right name will be made by cached_property above
        if hasattr(self, "log"):
            del self.log
```